src/task/builder.rs

```diff
@@ -0,0 +1,32 @@
+use super::pool;
+use super::JoinHandle;
+use crate::future::Future;
+use crate::io;
+
+/// Task builder that configures the settings of a new task.
+#[derive(Debug)]
+pub struct Builder {
+    pub(crate) name: Option<String>,
```

Comment:
Will you add lifecycle callbacks here? like `after_start`, `before_stop`?

Reply:
Do you mean lifecycles callbacks for tasks or for worker threads?

Reply:
Do you mean lifecycles callbacks for tasks or for worker threads?

Reply:
I think tokio has them for worker threads. What about doing it for `tasks` here?

Reply:
Yes, Tokio has them for worker threads. It doesn't make much sense for tasks because you can do:

```rust
let future = async move {
    // do stuff before
    let res = future.await;
    // do stuff after
    res
};
```